### PR TITLE
8266595: jdk/jfr/jcmd/TestJcmdDump.java with slowdebug bits fails with AttachNotSupportedException

### PR DESCRIPTION
**Pull request by egahlin**
Hi,

Could I have a review of a test fix that increases the timeout for attach. 

The test times out when running with slowdebug. I can reproduce the issue locally when running with the default 10 000 ms, but it works fine with 20 000 ms. I set the timeout to 50 000 ms, so attach at least have a chance to timeout, making it perhaps somewhat easier to analyze failures in the future (to rule timeout due to other reasons).

Thanks
Erik

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266595](https://bugs.openjdk.java.net/browse/JDK-8266595): jdk/jfr/jcmd/TestJcmdDump.java with slowdebug bits fails with AttachNotSupportedException

### Reviewers
 * [Mikhailo Seledtsov](https://openjdk.java.net/census#mseledtsov) (@mseledts - Committer) ⚠️ Review applies to 7ae464888644f5b9d7ff0d624d7075f863545d16
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 7ae464888644f5b9d7ff0d624d7075f863545d16

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/180/head:pull/180` \
`$ git checkout pull/180`

Update a local copy of the PR: \
`$ git checkout pull/180` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/180/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 180`

View PR using the GUI difftool: \
`$ git pr show -t 180`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/180.diff">https://git.openjdk.java.net/jdk17/pull/180.diff</a>

</details>

### CODE REVIEW

**Review by mseledts — APPROVED** [review]
Changes look good to me.

**Comment by egahlin** [comment]
> Changes look good to me.

Thanks Misha.

File: test/jdk/jdk/jfr/jcmd/TestJcmdDump.java [hunk 1]
```diff
@@ -105,8 +107,15 @@ private static void jfrDump(Boolean pathToGCRoots, String name, boolean disk, Pr
...
-            OutputAnalyzer output = JcmdHelper.jcmd(params);
+            List<String> params = buildParameters(pathToGCRoots, name, recording);
+            System.out.println(params);
+            OutputAnalyzer output = ProcessTools.executeProcess(new ProcessBuilder(params));
+            System.out.println("---------------- stdout ----------------");
+            System.out.println(output.getStdout());
+            System.out.println("---------------- stderr ----------------");
+            System.out.println(output.getStderr());
+            System.out.println("----------------------------------------");
+            System.out.println();
```
Comment (dholmes-ora) [inline test/jdk/jdk/jfr/jcmd/TestJcmdDump.java:118]: You can just call `output.reportDiagnosticSummary()` to get this.

File: test/jdk/jdk/jfr/jcmd/TestJcmdDump.java [hunk 2]
```diff
@@ -105,8 +107,15 @@ private static void jfrDump(Boolean pathToGCRoots, String name, boolean disk, Pr
             leakList.add(new Object[1000_0000]);
             System.gc(); // This will shorten time for object to be emitted.
             File recording = new File("TestJCMdDump.jfr");
-            String[] params = buildParameters(pathToGCRoots, name, recording);
-            OutputAnalyzer output = JcmdHelper.jcmd(params);
```
Comment (dholmes-ora) [inline test/jdk/jdk/jfr/jcmd/TestJcmdDump.java:109]: Not at all clear why you can't just add the additional timeout arg and still use the JcmdHelper ??

**Review by dholmes-ora — APPROVED** [review]
Hi Erik,

Not clear why you needed to do this the way you did. A query and a suggested change below.

Thanks,
David

File: test/jdk/jdk/jfr/jcmd/TestJcmdDump.java (hunk 2 above)
Comment (egahlin) [reply test/jdk/jdk/jfr/jcmd/TestJcmdDump.java:109]: I need to pass the the system property as the first argument. JcmdHelper uses the PidJcmdExecutor class in the test lib, which adds the PID as first argument, so I would need to rewrite both JcmdHelper and PidJcmdExecutor to support system properties functionality, which seemed overkill.

File: test/jdk/jdk/jfr/jcmd/TestJcmdDump.java (hunk 1 above)
Comment (egahlin) [reply test/jdk/jdk/jfr/jcmd/TestJcmdDump.java]: Updated to use output.reportDiagnosticSummary() instead.

**Pull request closed without merging**